### PR TITLE
[IMP] functions: add #DIV/0! error

### PR DESCRIPTION
**Pull request by LucasLefevre**
## Description:

description of this task, what is implemented and why it is implemented that way.

Task: : [3826335](https://www.odoo.com/web#id=3826335&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

**Comment by laa-odoo** [comment]
for consistency, I find it strange that this new `CellErrorType` does not have its class which extends `EvaluationError`.
But I understand why.

This makes me say that this is not the right approach. 
I have the impression that with what has been done before,
it appears that each type of error corresponds to a different part of the evaluation process.

In other words, it appears that a type of error cannot be raised on the same scale/file as another error ?

Here it is an error which has the particularity of being a calculation error, --> an error which can naturally be caught by javascript. So I wonder if we should not let the calculations be done naturally, and in case of a javascript error, look if the type of the js error corresponds to a division error and act accordingly

**Comment by LucasLefevre** [comment]
> it appears that a type of error cannot be raised on the same scale/file as another error ?

It may be a fact now, but why should it be something we want to enforce/keep ? Does it bring a value of any kind ?

> I wonder if we should not let the calculations be done naturally, and in case of a javascript error, look if the type of the js error corresponds to a division error and act accordingly

I don't think so because a division by zero is most probably something the developer should handle when implementing the function. Otherwise, it's a programming mistake/oversight and it should be fixed. What you propose would shadow those kind of mistakes.

**Comment by laa-odoo** [comment]
>  It may be a fact now, but why should it be something we want to enforce/keep ? Does it bring a value of any kind ?

I find that doing this allows, in a process as complex as the evaluation, to maintain ease of abstraction.
For example, I find it simple to say that in formula modules, we only have to worry about throwing simple errors (Evaluation error)

> I don't think so because a division by zero is most probably something the developer should handle when implementing the function. Otherwise, it's a programming mistake/oversight and it should be fixed. What you propose would shadow those kind of mistakes.

Maybe I'm wrong but it doesn't seem so at the moment ?
and even if we need to handle division errors in the sense that certain formulas mechanically return division errors, then they can use the mechanics of returning division errors and not throwing division errors ?

**Comment by laa-odoo** [comment]
It's just a thought

**Comment by LucasLefevre** [comment]
I must admit I don't really understand your point. The different kind of evaluation errors is only a matter of display. Using one over the other is just as good IMO. There's nothing to think about in my mind.

File: src/functions/module_statistical.ts [hunk 1]
```diff
@@ -80,7 +81,8 @@ function covariance(dataY: Arg, dataX: Arg, isSample: boolean): number {
 
   assert(
     () => count !== 0 && (!isSample || count !== 1),
-    _t("Evaluation of function [[FUNCTION_NAME]] caused a divide by zero error.")
+    _t("Evaluation of function [[FUNCTION_NAME]] caused a divide by zero error."),
```
Comment (hokolomopo) [inline src/functions/module_statistical.ts:84]: is the condition "(!isSample || count !== 1)" really a divide by zero error ?

File: src/functions/module_statistical.ts [hunk 2]
```diff
@@ -118,7 +120,8 @@ function variance(args: Arg[], isSample: boolean, textAs0: boolean, locale: Loca
...
```
Comment (hokolomopo) [inline src/functions/module_statistical.ts:123]: same

File: src/functions/module_statistical.ts (hunk 1 above)
Comment (LucasLefevre) [reply src/functions/module_statistical.ts:84]: :man_shrugging: not really but Google and/or Excel also behaves that way (I don't remember if I tried one or the other or both)

**Comment by rrahir** [comment]
@robodoo r+

**Review by LucasLefevre — COMMENTED** [review]
robodoo rebase-ff

**Pull request closed without merging**